### PR TITLE
GOBBLIN-1316: Copy field and schema level properties when switching namespace in a schema

### PR DESCRIPTION
**Pull request by sv2000**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1316

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
When a namespace override is applied to an input schema, the new schema does not currently contain the field and schema level properties of the original schema. 

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Unit test in AvroUtilsTest.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-utility/src/test/java/org/apache/gobblin/util/AvroUtilsTest.java [hunk 1]
```diff
@@ -51,9 +51,11 @@
 import org.codehaus.jackson.JsonNode;
 import org.codehaus.jackson.map.ObjectMapper;
 import org.codehaus.jackson.node.ArrayNode;
+import org.codehaus.jackson.node.JsonNodeFactory;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
+import com.beust.jcommander.internal.Lists;
```
Comment (htran1) [inline gobblin-utility/src/test/java/org/apache/gobblin/util/AvroUtilsTest.java:58]: Should this be the guava one instead?

File: gobblin-utility/src/test/java/org/apache/gobblin/util/AvroUtilsTest.java [hunk 2]
```diff
@@ -235,8 +237,21 @@ public void testSwitchNamespace() {
     String originalNamespace = "originalNamespace";
     String originalName = "originalName";
     String newNamespace = "newNamespace";
-    Schema schema = SchemaBuilder.builder(originalNamespace).record(originalName).fields().
-        requiredDouble("double").optionalFloat("float").endRecord();
+    //Schema schema = SchemaBuilder.builder(originalNamespace).record(originalName).fields().
```
Comment (htran1) [inline gobblin-utility/src/test/java/org/apache/gobblin/util/AvroUtilsTest.java:240]: Remove these lines?

**Review by htran1 — CHANGES_REQUESTED** [review]

**Review by htran1 — APPROVED** [review]
+1

File: gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java [hunk 3]
```diff
@@ -781,11 +781,19 @@ public static Schema switchNamespace(Schema schema, Map<String, String> namespac
           for (Schema.Field oldField : schema.getFields()) {
             Field newField = new Field(oldField.name(), switchNamespace(oldField.schema(), namespaceOverride), oldField.doc(),
                 oldField.defaultValue(), oldField.order());
+            // Copy field level properties
+            for (Map.Entry<String, JsonNode> prop : oldField.getJsonProps().entrySet()) {
```
Comment (shirshanka) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java:785]: Use the already defined copyFieldProperties method?

File: gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java [hunk 4]
```diff
@@ -781,11 +781,19 @@ public static Schema switchNamespace(Schema schema, Map<String, String> namespac
...
+              newField.addProp(prop.getKey(), prop.getValue());
+            }
             newFields.add(newField);
           }
         }
         newSchema = Schema.createRecord(schema.getName(), schema.getDoc(), newNamespace,
             schema.isError());
+        // Copy schema level properties
```
Comment (shirshanka) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java:793]: Use the already defined copyProperties method?

**Review by shirshanka — CHANGES_REQUESTED** [review]
It's a small change, but can be shortened further I think!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request closed without merging**